File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/bpm/notifications/UserMetadataTransitionNotificationModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.notifications;


public class UserMetadataTransitionNotificationModel extends TransitionNotificationModel {

	private static final long serialVersionUID = 1L;
	
	public static final String PROPERTY_METADATA_NAME = "metadataName";
	
	public UserMetadataTransitionNotificationModel() {
		super(TransitionNotificationModelType.METADATA);
	}

	public String getMetadataName() {
		return get(PROPERTY_METADATA_NAME);
	}
	
	public void setMetadataName(String metadataName) {
		set(PROPERTY_METADATA_NAME, metadataName);
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/bpm/notifications/user-metadata-transition-notification.model.ts
import { TransitionNotificationModelType } from "./transition-notification-model-type";
import { TransitionNotificationModel } from "./transition-notification.model";
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class UserMetadataTransitionNotificationModel extends TransitionNotificationModel {

	public constructor() {
		super(TransitionNotificationModelType.METADATA);
	}

	@JsonProperty("metadataName", String)
	public metadataName: string;
}